pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -246,8 +251,12 @@ private void manageSegmentLineageCleanupForTable(String tableNameWithType) {
             .writeSegmentLineage(_pinotHelixResourceManager.getPropertyStore(), segmentLineage, expectedVersion)) {
           // Delete segments based on the segment lineage
           _pinotHelixResourceManager.deleteSegments(tableNameWithType, segmentsToDelete);
+          LOGGER.info("Finished cleaning up segment lineage for table: {} in {}ms", tableNameWithType,
+              (System.currentTimeMillis() - cleanupStartTime));
           return true;
         } else {
+          LOGGER.info("Failed to write segment lineage back when cleaning up segment lineage for table: {}",
```

Comment:
This could be a `warn`? With current logic, this usually should not happen. This can only happen when `startSegmentReplace` or `endSegmentReplace` is called at the same time when retention process is running?

Reply:
Changed to `warn`. Right, this is to ensure no two threads try to update the same ZNode at the same time.